ex1/ex1.c

```diff
@@ -2,13 +2,50 @@
 // (e.g., x) and set its value to something (e.g., 100). What value is the variable in the child process?
 // What happens to the variable when both the child and parent change the value of x?
 
+// ~~ Output ~~
+// $ ./ex1
+// Pre-fork x: 0, from PID: 2870
+// Parent PID: 2870
+// Parent before setting x: 0
+// Parent after setting x: 100
+// Post-fork x: 100, from PID: 2870
+// Child PID: 2871
+// Child before setting x: 0
+// Child after setting x: 50
+// Post-fork x: 50, from PID: 2871
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
 int main(void)
 {
     // Your code here
+    int x = 0;
+    printf("Pre-fork x: %i, from PID: %i\n", x, (int)getpid());
+
+    int fark = fork();
+    if (fark < 0)
+    {
+        fprintf(stderr, "Failed to fork child process\n");
+        exit(1);
+    }
+    else if (fark == 0)
+    {
+        printf("Child PID: %i\n", (int)getpid());
+        printf("Child before setting x: %i\n", x);
+        x = 50;
+        printf("Child after setting x: %i\n", x);
+    }
+    else
+    {
+        printf("Parent PID: %i\n", (int)getpid());
+        printf("Parent before setting x: %i\n", x);
+        x = 100;
+        printf("Parent after setting x: %i\n", x);
+    }
+
+    printf("Post-fork x: %i, from PID: %i\n", x, (int)getpid());
```

Comment:
nice going with this statement here. it makes super clear that these are separate processes entirely